### ISSUE TITLE
Connect to MySql with PhpMyAdmin in production fails in production, not allowed to connect to this Mysql

### ISSUE THREAD

**Issue by Shmp** [laradock/laradock#379]
I'm trying to connect to my mysql database with phpmyadmin. Because it is not recommended to expose ports on the mysql database on production, I can't connect to de mysql container. I can only do this when i'm exposing port 3306.

What are your solutions?

**Comment by philtrep** [comment]
@Shmp 
There are 2 instructions in docker-compose that manage ports: `ports` and `expose`.

`expose` gives access to that port to other containers:
```
 mysql:
        [...]
        expose:
                - "3306"
```
That would give access to `mysql:3306` to PHPMyAdmin, workspace, etc.

`ports` forwards the specified port to the host machine. Forwarding ports to the host does something bad by default, it automatically publishes the affected port from the host:
```
mysql:
        [...]
        ports:
                - "3306:3306"
```

This as the additional effect of giving the host access to the database (through `localhost:3306` not `mysql:3306` since the port is being forwarded). The problem here is that docker will by default manage the iptables on the host machine, you would therefore be allowed to connect remotely to the database.

**TLDR** Use `expose` instead of `ports` in your docker-compose file.

**Comment by Shmp** [comment]
@philtrep Thanks, will try this right away!

**Comment by Shmp** [comment]
@philtrep Sweet! Works like a charm!

**Comment by Shmp** [comment]
Got another problem now:
` #1130 - Host '172.18.0.7' is not allowed to connect to this MySQL server`
`mysqli_real_connect(): (HY000/1130): Host '172.18.0.7' is not allowed to connect to this MySQL server`

deleted data, rebuilded all containers serveral times, added --bind-address=0.0.0.0

can't connect to the database...

**Comment by philtrep** [comment]
@Shmp 

Try flushing the privileges and recreating the user with a more permissive host (`%`). If this doesn't work, do a dump, delete all the `data/mysql`, recreate the container and import the dump in your fresh database.

**Comment by Shmp** [comment]
@philtrep I've read a few post on other project and it seems that mysql default doesn't allow connections from another host. Is there a way to edit some config or compose file to create a rule to allow connections?

**Comment by philtrep** [comment]
@Shmp try changing the `my.cnf` to:
```
[mysqld]
bind-address = 0.0.0.0
```

You'll need to rebuild your container but I think that might do the trick...
